Automattic/wp-calypso
# [Blocked] Starter page templates: Remove iframe from preview

A number of reported issues with the preview in the page layout picker can be resolved by removing the iframe from the preview. Certain blocks (such as masonry, or maps blocks) require JavaScript to complete their layouts in the case of masonry, or to perform API calls in the case of a map block. By removing the iframe, the preview has access to the JavaScript already loaded on the page, so we don't need to add any additional hacks to get these blocks working, and it also inches the preview a bit closer to how the preview works in the pattern inserter.

Issues resolved by this change:

* Fixes: #48889 (Contact form not rendered correctly)
* Fixes: #48883 (Issue with rendering map block in preview)
* Fixes part of: #48881 (Rendering issue with galleries in thumbnails and previews)

#### Changes proposed in this Pull Request

* Remove the iframe from the starter page templates preview, and associated iframe methods
* Tweak styling and existing features to ensure there aren't any regressions

#### Screenshots

Note in the following that the map works in the preview, and that the masonry layout renders:

![spt-remove-iframe](https://user-images.githubusercontent.com/14988353/105305338-18309000-5c10-11eb-8157-b8e4af402dfb.gif)

#### Testing instructions



* Sandbox the diff associated with this PR and a test site
* In your test site go to add a new page, and test that:
  * The masonry layout (the first Gallery layout) renders correctly in the preview (see #48881 for more details) and note that this does not resolve the thumbnail on the button
  * The map renders in the preview for the second last Contact layout (see https://github.com/automattic/wp-calypso/issues/48883 for more details
  * Test that switching layouts updates the post title in the preview
  * Test that resizing the window also re-scales the preview after a short while (300ms — the debounce timeout)
* Test that, in general, the preview looks better with this change applied than without it — there might still be some oddities, but the objective here is to ensure that it's an improvement
* Test across multiple browsers / devices to check for regressions

CC: @Automattic/ganon — just copying you in, in case you get a chance to review since you're also doing work on this area 🙂
